smart_open/s3.py

```diff
@@ -658,6 +674,130 @@ def __repr__(self):
         )
 
 
+class BufferedSinglePartOutputBase(io.BufferedIOBase):
+    """Writes bytes to S3 using the single part API.
+
+    Implements the io.BufferedIOBase interface of the standard library."""
```

Comment:
You should mention that this class buffers all of its input, and only writes to S3 when `close` is called.